### PR TITLE
Query architecture CUDA code was compiled for with nvc++

### PR DESCRIPTION
**Pull request by dalg24**
Update check that code was compiled for the right architecture to support nvc++

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp [hunk 1]
```diff
@@ -107,11 +107,15 @@ namespace Impl {
 namespace {
 
 __global__ void query_cuda_kernel_arch(int *d_arch) {
+#if defined(KOKKOS_COMPILER_NVHPC) && defined(_NVHPC_CUDA)
+  *d_arch = __builtin_current_device_sm() * 10;
```
[the remark this answers is not in the document]
Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:111]: > Is this documented anywhere (source code or the like)?

Not yet

> Does multiplying by 10 mean that we can't detect, e.g. Volta70, vs. Volta72?

No that would work, sm is 70 vs 72 and `__CUDA_ARCH__` would be 700 vs 720

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp [hunk 2]
```diff
@@ -107,11 +107,15 @@ namespace Impl {
 namespace {
 
 __global__ void query_cuda_kernel_arch(int *d_arch) {
+#if defined(KOKKOS_COMPILER_NVHPC) && defined(_NVHPC_CUDA)
```
Comment (masterleinad) [inline core/src/Cuda/Kokkos_Cuda_Instance.cpp:110]: Isn't 
```suggestion
#ifdef _NVHPC_CUDA
```
sufficient?

**Review by Rombur — APPROVED** [review]

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp (hunk 2 above)
Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:110]: I suppose so.  Are you asking me to change?

**Review by crtrott — APPROVED** [review]

File: core/src/Cuda/Kokkos_Cuda_Instance.cpp (hunk 2 above)
Comment (masterleinad) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:110]: We are not using `_NVHPC_CUDA` anywhere else if I'm not mistaken. I think we should only be consistent.

Comment (dalg24) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:110]: That is the first time we use it indeed.
>  I think we should only be consistent.

No one is going to argue with that.  I don't understand what you mean though.  Are you saying you would like me to apply your suggestion?  (I have no objection)

Comment (masterleinad) [reply core/src/Cuda/Kokkos_Cuda_Instance.cpp:110]: Yeah, I would prefer to change it (to be closer to what we do below for CUDA without NVHPC).

**Review by masterleinad — APPROVED** [review]

**Pull request merged**